### PR TITLE
Refactor: Style quantity input dynamically

### LINKED ISSUE #6029: Mini Cart: Style doesn't apply to quantity input
## To reproduce

1. Edit Mini Cart Template part.
2. Set the text, link, and background color for the Mini Cart Contents block.
3. See the color doesn't apply to the quantity input in the editor.
4. Save. See the issue on the front end.

## Expected behavior

Quantity input should be styled as other elements.

## Screenshots

image

## Additional context

Thank @nerrad to report this issue!

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #6029, replaces #6087

This PR refactors the style of input quantity used by Cart and Mini Cart blocks to make it dynamic from the CSS side. This PR tries to replicate as close as possible the original style of the quantity input but there are some parts changed to make it dynamic.

### Screenshots

_Notes: The original style is on the right. The style from this PR is on the left._

Default state (Cart block):
image

The new border is darker and bolder because it works better for colors other than black.

Focused input: (Mini Cart block)
image

Focused button: (Mini Cart block)
image

Dynamic style: (Mini Cart block)
image

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Checkout this PR and build the assets.
2. Test the quantity input of the Cart block, see no issue, see style for normal, hover, focused, and disabled state of input and button look good.
3. For Mini Cart, set the text color and background of the Mini Cart Contents block. See the input style changes dynamically.
### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [ ] Same as above, or
* [ ] See steps below.

### CODE REVIEW

File: assets/js/base/components/quantity-selector/style.scss
Comment (gigitux) [inline assets/js/base/components/quantity-selector/style.scss]: I know that it is not your change, but I guess that we could use [`appearance`](https://caniuse.com/css-appearance). What do you think?

**Review by gigitux — APPROVED** [review]
Just a little comment about not using the `moz` prefix.

I noticed that the hover effect works even if the button is disabled. Maybe we should disable the hover effect for the disabled buttons? What do you think?

image

Besides this, LGTM! I'm pre-approving the PR! Thanks for working on this!

**Comment by dinhtungdu** [comment]
@gigitux Thanks for the review! I updated my PR per your suggestions.

Because this changes the design of the input, I'm requesting an additional review from @vivialice. Please take a closer look at the focus style of input and button, I think we should make them clearer.

I also remove the `.has-dark-controls` related style because we don't need them anymore.

**Comment by dinhtungdu** [comment]
I'm trying a different approach on [fix/6029-3](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/fix/6029-3). I use pseudo-selectors to style the quantity wrapper and increase the contrast of the focus state border for buttons and input. Using the color picker to check the color of the border, it passes the Graphical Objects and User Interface Components WCAG AA: https://webaim.org/resources/contrastchecker/?fcolor=8D8D8D&bcolor=FFFFFF.

With approach on [fix/6029-3](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/fix/6029-3), here is how the quantity input look inside the Cart block.

Default style with focus input:
image

Dark mode:
image

Dark mode with focused input:
image

**Comment by Aljullu** [comment]
@dinhtungdu what's the status of this PR? Does it need another review or are you planning on creating another PR based on `fix/6029-3`?

**Comment by dinhtungdu** [comment]
> @dinhtungdu what's the status of this PR? Does it need another review or are you planning on creating another PR based on `fix/6029-3`?

@Aljullu Because this changes the design so I'm waiting for approval from @vivialice. Also, I'm waiting for the input on `fix/6029-3` before merging it to this PR.

**Comment by dinhtungdu** [comment]
> Also, I'm waiting for the input on fix/6029-3 before merging it to this PR.

@Aljullu @gigitux I decided to merge `fix/6029-3` in [e854f0b](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6167/commits/e854f0b3dddb22da80a85bb2bcf74adf35e6ff2c).  Please give this PR another review because the quantity input wrapper now uses pseudo selectors to styling the border.

**Review by gigitux — APPROVED** [review]
LGTM! 🎉

**Comment by dinhtungdu** [comment]
The failed JS linting workflow is due to https://github.com/ataylorme/eslint-annotate-action/issues/39. Failed E2E tests are unrelated to this PR. So I'm merging this PR.

**Pull request merged**
